src/main/java/mekanism/common/tile/component/TileComponentEjector.java

```diff
@@ -139,52 +123,53 @@ private void ejectFluid(Set<EnumFacing> outputSides, FluidTank tank) {
         return outputSides;
     }
 
-    public void outputItems() {
-        if (!getEjecting(TransmissionType.ITEM) || tileEntity.getWorld().isRemote) {
+    private void outputItems() {
+        SideData data = sideData.get(TransmissionType.ITEM);
+        if (data == null || !getEjecting(TransmissionType.ITEM)) {
             return;
         }
-
-        SideData data = sideData.get(TransmissionType.ITEM);
-        Set<EnumFacing> outputSides = getOutputSides(TransmissionType.ITEM, data);
-        for (int index = 0; index < data.availableSlots.length; index++) {
-            int slotID = data.availableSlots[index];
-            if (tileEntity.getStackInSlot(slotID).isEmpty()) {
-                continue;
-            }
-
-            ItemStack stack = tileEntity.getStackInSlot(slotID);
-            Coord4D tileCoord = Coord4D.get(tileEntity);
-            Set<EnumFacing> outputs = getTrackedOutputs(TransmissionType.ITEM, index, outputSides);
-            for (EnumFacing side : outputs) {
-                TileEntity tile = tileCoord.offset(side).getTileEntity(tileEntity.getWorld());
-                int prevCount = stack.getCount();
-
-                ILogisticalTransporter capability = CapabilityUtils.getCapability(tile, Capabilities.LOGISTICAL_TRANSPORTER_CAPABILITY, side.getOpposite());
-                TransitRequest transitRequest = TransitRequest.getFromStack(stack.copy());
-                TransitResponse response;
-                if (capability == null) {
-                    response = InventoryUtils.putStackInInventory(tile, transitRequest, side, false);
-                } else {
-                    response = TransporterUtils.insert(tileEntity, capability, transitRequest, outputColor, true, 0);
-                }
-                if (!response.isEmpty()) {
-                    stack.shrink(response.getSendingAmount());
-                }
-
-                if (stack.isEmpty() || prevCount != stack.getCount()) {
-                    trackers.get(TransmissionType.ITEM)[index] = side.ordinal();
-                }
-                if (stack.isEmpty()) {
+        Set<EnumFacing> outputs = getOutputSides(TransmissionType.ITEM, data);
+        TransitRequest ejectMap = null;
+        for (EnumFacing side : outputs) {
+            if (ejectMap == null) {
+                ejectMap = getEjectItemMap(data);
+                if (ejectMap.isEmpty()) {
                     break;
                 }
             }
+            BlockPos offsetPos = tileEntity.getPos().offset(side);
+            TileEntity tile = tileEntity.getWorld().isBlockLoaded(offsetPos) ? tileEntity.getWorld().getTileEntity(offsetPos) : null;
```

Comment:
If 1.12 is merged back into this, then we can do `tile = MekanismUtils.getTile` to check for the isBlockLoaded in an easier way to read